### PR TITLE
UI fixes for File transfers ( PULL AFTER #917 )

### PR DESCRIPTION
**Pull request by GrayHatter**
Please pull #917  first!

@tsudoko @muchweb I've broken some interface coloring, can you guys take a look and see if you can clean it up a bit?

### CODE REVIEW

File: file_transfers.h [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+} FILE_TRANSFER;
+
+typedef struct {
+    uint64_t size_transferred;
+    uint32_t speed;
+} FILE_PROGRESS;
+
+/* Callback for file transfer changes made by friend */
+void file_transfer_local_control(Tox *tox, uint32_t friend_number, uint32_t file_number, TOX_FILE_CONTROL control);
+static void file_transfer_callback_control(Tox *tox, uint32_t friend_number, uint32_t file_number, TOX_FILE_CONTROL control, void *UNUSED(userdata));
```
Comment (irungentoo) [inline file_transfers.h]: static functions in a .h file?

Comment (GrayHatter) [inline file_transfers.h]: I wanted to put all the internal functions at the bottom of the .c while still using them above. I can re order the functions, and remove the prototypes if you'd like.

I did in this was so I can just write the documentation in the .h and not miss any functions...

**Comment by irungentoo** [comment]
```
#0  0x0000003d7e47848c in fseeko64 () from /lib64/libc.so.6
#1  0x000000000053276f in outgoing_file_callback_chunk (tox=0x61f00001ea80, friend_number=28, file_number=0, position=72663, 
    length=1371, user_data=0x0) at file_transfers.c:233
#2  0x0000000000f2f688 in do_reqchunk_filecb (m=<optimized out>, friendnumber=<optimized out>) at ../toxcore/Messenger.c:1487
#3  do_friends (m=<optimized out>) at ../toxcore/Messenger.c:2227
#4  0x0000000000f32b17 in do_messenger (m=0x61f00001ea80) at ../toxcore/Messenger.c:2305
#5  0x0000000000f3e6d9 in tox_iterate (tox=0x61f00001ea80) at ../toxcore/tox.c:353
#6  0x0000000000652220 in tox_thread (UNUSED_args=<optimized out>) at tox.c:624
#7  0x0000003d7f0091da in start_thread () from /lib64/libpthread.so.0
#8  0x0000003d7e4edbed in clone () from /lib64/libc.so.6
```

crashes because the file pointer passed to fseeko is NULL.

**Comment by GrayHatter** [comment]
@irungentoo the segfault should be fixed in #920

**Pull request closed without merging**